openedx/core/djangoapps/credit/api.py

```diff
@@ -0,0 +1,102 @@
+""" Contains the APIs for course credit requirements """
+
+from .exceptions import InvalidCreditRequirements
+from .models import CreditCourse, CreditRequirement
+
+
+def set_credit_requirements(course_key, requirements):
+    """ Add requirements to given course
+
+    Args:
+        course_key(CourseKey): The identifier for course
+        requirements(list): List of requirements to be added
+
+    Example:
+        >>> set_credit_requirements(
+                "course-v1-ASUx-DemoX-1T2015",
+                [
+                    {
+                        "namespace": "verification",
+                        "name": "verification",
+                        "criteria": {},
+                    },
+                    {
+                        "namespace": "reverification",
+                        "name": "midterm",
+                        "criteria": {},
+                    },
+                    {
+                        "namespace": "proctored_exam",
+                        "name": "final",
+                        "criteria": {},
+                    },
+                    {
+                        "namespace": "grade",
+                        "name": "grade",
+                        "criteria": {"min_grade": 0.8},
+                    },
+                ])
+
+    Raises:
+        InvalidCreditRequirements
+
+    Returns:
+        None
+    """
+
+    try:
+        credit_course = CreditCourse.get_credit_course(course_key=course_key)
+        for requirement in requirements:
+            CreditRequirement.add_course_requirement(credit_course, requirement)
```

Comment:
This also needs to check if there are existing requirements that are no longer valid (for example if an ICRV block got deleted) and mark those as disabled.